testing/acceptance_test.py

```diff
@@ -1179,3 +1179,23 @@ def test_fixture_mock_integration(testdir):
 def test_usage_error_code(testdir):
     result = testdir.runpytest("-unknown-option-")
     assert result.ret == EXIT_USAGEERROR
+
+
+@pytest.mark.skipif(
+    sys.version_info[:2] < (3, 5), reason="async def syntax python 3.5+ only"
+)
+@pytest.mark.filterwarnings("default")
+def test_warn_on_async_function(testdir):
+    testdir.makepyfile(
+        test_async="""
+        async def test_1():
+            pass
+    """
+    )
+    result = testdir.runpytest()
+    result.stdout.fnmatch_lines(
+        [
+            "*test function test_async.py::test_1 is a coroutine*",
+            "*1 passed, 1 warnings in*",
```

Comment:
Should it be counted as "passed" after all?

Reply:
Hmmm you are right.

I suppose the ideal behavior would be to not collect that test at all, but that's not possible, after all the async plugins need that collected.

Should we skip or xfail it, perhaps?

Reply:
> Should we skip or xfail it, perhaps?

Skipping sounds good to me - but there should probably only be a single skip message (grouped) with `-ra` - might be the case automatically, when skipping from the same location here, but should have a test.

Reply:
When we skip, should we still show the warning? The warning is a lot more verbose, which doesn't fit with "reason" which is usually a single line...

Reply:
IMHO yes.  The skip message could state to look at the warning for more details.

Also, having a subclass of PytestWarning for this might make sense, for easier ignoring?!

Reply:
Done, here's the full output of the test:

```
======================================== warnings summary =========================================
test_async.py::test_1
test_async.py::test_2
  C:\pytest\src\_pytest\python.py:166: PytestWarning: Coroutine functions are not natively supported and have been skipped.
  You need to install a suitable plugin for your async framework, for example:
    - pytest-asyncio
    - pytest-trio
    - pytest-tornasync
    warnings.warn(PytestWarning(msg.format(pyfuncitem.nodeid)))

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============================== 2 skipped, 2 warnings in 0.02 seconds ==============================
```

> Also, having a subclass of PytestWarning for this might make sense, for easier ignoring?!

Indeed, but let's wait to see if there's user requests for that. If we go down that route, it might make sense to make a subclass for each warning emitted by pytest.